dbtvault-dev/macros/tables/eff_sat.sql

```diff
@@ -21,105 +21,109 @@
-    SELECT DISTINCT
-        {{ dbtvault.alias_all(source_cols, 'stage') }}
-    FROM stage_slice AS stage
-    LEFT JOIN latest_open_eff AS e
-    ON stage.{{ src_pk }} = e.{{ src_pk }}
-    WHERE e.{{ src_pk }} IS NULL
-    AND {{ dbtvault.multikey(src_dfk, prefix='stage', condition='IS NOT NULL') }}
-    AND {{ dbtvault.multikey(src_sfk, prefix='stage', condition='IS NOT NULL') }}
+{# Selecting the closed records of the most recent records for each link hashkey -#}
+latest_closed AS (
+    SELECT {{ dbtvault.alias_all(source_cols, 'd') }}
+    FROM latest_records AS d
+    WHERE TO_DATE(d.{{ src_end_date }}) != TO_DATE('9999-12-31')
 ),
-{%- if is_auto_end_dating %}
 
-links_to_end_date AS (
-    SELECT a.*
-    FROM latest_open_eff AS a
-    LEFT JOIN stage_slice AS b
-    ON {{ dbtvault.multikey(src_dfk, prefix=['a', 'b'], condition='=') }}
-    WHERE {{ dbtvault.multikey(src_sfk, prefix='b', condition='IS NULL', operator='OR') }}
-    OR {{ dbtvault.multikey(src_sfk, prefix=['a', 'b'], condition='<>', operator='OR') }}
+{# Identifying the completely new link relationships to be opened in eff sat -#}
+new_open_records AS (
+    SELECT DISTINCT
+        {{ dbtvault.alias_all(source_cols, 'f') }}
+    FROM source_data AS f
+    LEFT JOIN latest_records AS lr
+    ON f.{{ src_pk }} = lr.{{ src_pk }}
+    WHERE lr.{{ src_pk }} IS NULL
 ),
 
-new_end_dated_records AS (
+{# Identifying the currently closed link relationships to be reopened in eff sat -#}
+new_reopened_records AS (
     SELECT DISTINCT
-        h.{{ src_pk }},
-        {{ dbtvault.alias_all(fk_cols, 'g') }},
-        h.{{ src_eff }} AS {{ src_start_date }},
-        h.{{ src_source }}
-    FROM latest_open_eff AS h
-    INNER JOIN links_to_end_date AS g
-    ON g.{{ src_pk }} = h.{{ src_pk }}
+        lc.{{ src_pk }}
+        ,{{ dbtvault.alias_all(fk_cols, 'lc') }}
+        ,lc.{{ src_start_date }} AS {{ src_start_date }}
+        ,g.{{ src_end_date }} AS {{ src_end_date }}
+        ,g.{{ src_eff }} AS {{ src_eff }}
+        ,g.{{ src_ldts }}
+        ,g.{{ src_source }}
+    FROM source_data AS g
+    INNER JOIN latest_closed lc
+    ON g.{{ src_pk }} = lc.{{ src_pk }}
 ),
 
-amended_end_dated_records AS (
+{%- if is_auto_end_dating %}
+
+{# Creating the closing records -#}
+{# Identifying the currently open relationships that need to be closed due to change in SFK(s) -#}
+new_closed_records AS (
     SELECT DISTINCT
-        a.{{ src_pk }},
-        {{ dbtvault.alias_all(fk_cols, 'a') }},
-        a.{{ src_start_date }},
-        stage.{{ src_eff }} AS END_DATE, stage.{{ src_eff }}, stage.{{ src_ldts }},
-        a.{{ src_source }}
-    FROM new_end_dated_records AS a
-    INNER JOIN stage_slice AS stage
-    ON {{ dbtvault.multikey(src_dfk, prefix=['stage', 'a'], condition='=') }}
-    WHERE {{ dbtvault.multikey(src_sfk, prefix='stage', condition='IS NOT NULL') }}
-    AND {{ dbtvault.multikey(src_dfk, prefix='stage', condition='IS NOT NULL') }}
+        lo.{{ src_pk }}
```

Comment:
Please fix formatting. All SQL code should have trailing commas instead of leading.

Reply:
Resolved